### ISSUE TITLE
Add support of signature algorithms (EdDSA, PS256, PS384, PS512, HS384, HS512) in quarkus-oidc

### ISSUE THREAD

**Issue by Eng-Fouad** [quarkusio/quarkus#33444]
### Description

https://github.com/quarkusio/quarkus/blob/aad9ce5afbba406a56528ca962b2f352344a2c3c/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcProvider.java#L45-L54

The following signature algorithms are missing:

- EdDSA
- PS256
- PS384
- PS512
- HS384
- HS512

**Comment by sberyozkin** [comment]
@Eng-Fouad as far as `EdDSA` is concerned, sure, it has been a plan since some providers like Curity support it.
`PS256, PS384, PS512` - likewise.
But `HS384, HS512` (and `HS256`) is unlikely to happen - where would quarkus get the secret keys from to verify signatures ? Note `HS256` signing is only supported for the internally generated ID tokens when providers like `GitHub` do not provide ID tokens at all

**Comment by sberyozkin** [comment]
Sure, lers get those added

**Comment by Eng-Fouad** [comment]
Adding `PS256`, `PS384`, and `PS512` is just by modifying:

https://github.com/quarkusio/quarkus/blob/aad9ce5afbba406a56528ca962b2f352344a2c3c/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcProvider.java#L45-L54

to:

```java
private static final String[] ASYMMETRIC_SUPPORTED_ALGORITHMS = new String[] { SignatureAlgorithm.RS256.getAlgorithm(), 
         SignatureAlgorithm.RS384.getAlgorithm(), 
         SignatureAlgorithm.RS512.getAlgorithm(), 
         SignatureAlgorithm.ES256.getAlgorithm(), 
         SignatureAlgorithm.ES384.getAlgorithm(), 
         SignatureAlgorithm.ES512.getAlgorithm(),
         SignatureAlgorithm.PS256.getAlgorithm(),
         SignatureAlgorithm.PS384.getAlgorithm(),
         SignatureAlgorithm.PS512.getAlgorithm() }; 
 private static final AlgorithmConstraints ASYMMETRIC_ALGORITHM_CONSTRAINTS = new AlgorithmConstraints( 
         AlgorithmConstraints.ConstraintType.PERMIT, ASYMMETRIC_SUPPORTED_ALGORITHMS); 
 private static final AlgorithmConstraints SYMMETRIC_ALGORITHM_CONSTRAINTS = new AlgorithmConstraints( 
         AlgorithmConstraints.ConstraintType.PERMIT, SignatureAlgorithm.HS256.getAlgorithm()); 
```

as it's supported by JDK11+ by default. Is that correct?


Regarding `EdDSA`, it needs JDK15+ or different provider like Bouncy Castle.

**Comment by sberyozkin** [comment]
@Eng-Fouad Thanks, you can add `EdDSA`, and we actually test it in `smallrye-jwt-build` if its Java 17. I haven't tested with BC, but it should work too. But adding a String there won't harm, if users do need `EdDSA` then they will need to run either in Java 17+ or use BouncyCastle.

**Comment by Eng-Fouad** [comment]
Resolved by #33462
